doc/releases/release-notes-4.1.rst

```diff
@@ -985,3 +985,14 @@ Other notable changes
   by excluding the test scenario name prefix which is the same as the parent test suite id.
 
 * Added support for HTTP PUT/PATCH/DELETE methods for HTTP server dynamic resources.
+
+* Driver API structures are now available through iterable sections and a new
+  :c:macro:`DEVICE_API_IS` macro has been introduced to allow to check if a device supports a
+  given API. Many shell commands now use this to provide "smarter" auto-completion and only list
+  relevant devices when they expect a device argument.
```

Comment:
maybe instead of smarter something like "api aware"? and "supported devices" instead of "relevant devices"?

Reply:
"API aware" sounds really weird to me. I basically tried to word the whole thing from the POV of the end-user who, for better or worse, might not (yet?) no a lot about device model and stuff -- all they care about if that if a command is apparently expecting a sensor, code completion will now be restricted to _relevant_ entries, i.e. sensors in this case.

Reply:
mmh ok, then how about "compatible devices" rather than "relevant devices"? in the end it's about devices with the correct api for the command